src/french.js

```diff
@@ -1,68 +1,434 @@
+        gender: 'female/male/unknown',
+          undefined subject: ils (english: they), nous (english: we) , vous (you, adressing a group of persons), the people -> is conjugated as male, when we don't know who 'they', 'nous', 'vous' represent (male or female)
+          unknown: only for 'on' with the meaning 'it'
+          ! 'on' and objects 'can' also have a gender !
+      }
+  */
   conjugate (word, info) {
-    // Format for info: {tense: 'present', formal: 'true/false', gender: 'female/male/undefined', singular: 'true/false/undefined', person: '1'/'2'/'3'}
-    //TODO define 3 verb group types
+    switch (info.mood) {
+      case 'Indicatif':
+        switch (info.tense) {
+          case 'Présent':
+            return this.doIndicatifPresent(word, info);
+            break;
+          case 'Passé Composé':
+            break;
+          case 'Imparfait':
+            break;
+          case 'Plus-que-parfait':
+            break;
+          case 'Passé Simple':
+            break;
+          case 'Passé Antérieur':
+            break;
+          case 'Futur Simple':
+            break;
+          case 'Futur Antérieur':
+            break;
+          default:
+            return 'Indicatif Tense not found';
+        }
+        break;
+      case 'Infinitif':
+        switch (info.tense) {
+          case 'Présent':
+           return word;
+           break;
+          case 'Passé':
+            break;
+          default:
+            return 'Infinitif Tense not found';
+        }
+        break;
+      case 'Participe':
+        switch (info.tense) {
+          case 'Présent':
+           break;
+          case 'Passé':
+            break;
+          default:
+            return 'Participe Tense not found';
+        }
+        break;
+      case 'Subjonctif':
+        switch (info.tense) {
+          case 'Présent':
+            break;
+          case 'Passé':
+            break;
+          case 'Imparfait':
+            break;
+          case 'Plus-que-parfait':
+            break;
+          default:
+            return 'Subjonctif Tense not found';
+        }
+        break;
+      case 'Impératif':
+        switch (info.tense) {
+          case 'Présent':
+            break;
+          case 'Passé':
+            break;
+          default:
+            return 'Impératif Tense not found';
+        }
+        break;
+      case 'Conditionnel':
+        switch (info.tense) {
+          case 'Présent':
+            break;
+          case 'Passé 1':
+            break;
+          case 'Passé 2':
+            break;
+          default:
+          return 'Conditionnel Tense not found';
+        }
+        break;
+      default:
+        return 'Mood not found';
+    }
+  }
 
-   switch (info.tense) {
-     case 'present':
-       return this.doPresent(word, info);
-       break;
-     default:
-       return 'Could not find any rules for conjugation';
-   }
+  findSubject (info) {
```

Comment:
this is retrieving the subject. The standard naming convention for retrieval is get. I would call this `getSubject`